### PR TITLE
Add GitHub validation task (X-Signature and repository URL)

### PR DESCRIPTION
**Pull request by a-roberts**
# Changes

Adds a validation task with accompanying docs, testing the repository URL part next. This is for https://github.com/tektoncd/experimental/issues/245

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by a-roberts** [comment]
I have the validate task working checking the repository URL as well as the signature now, would like to get the task/ingress PRs in first so I can accomodate the readme accordingly 😄 

@dibbles FYI

File: webhooks-extension/config/triggers_prototype/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## GitHub validation task
+
+The provided validation task is to be used in combination with an EventListener passing in EventOrigin, Github-Secret and Github-Secret-Key as parameters.
+
+The secret key should be `secretToken` if using a secret created through the Tekton Webhooks Extension.
+
+Two steps of validation can be performed.
+
+1) Mandatory every time: that the secret for the webhook matches with the known secret on this Kubernetes cluster (used by the validate task)
+2) Optional if you provide the EventOrigin oarameter: that the repository URL for the incoming webhook matches what the EventListener is listening for (done on a "per Trigger" basis)
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/README.md]: oarameter ????

Comment (dibbles) [reply webhooks-extension/config/triggers_prototype/README.md]: Optional if you provide the EventOrigin parameter: that the repository URL for the incoming webhook matches what the parameter specified in the EventListener (done on a "per Trigger" basis)

File: webhooks-extension/config/triggers_prototype/validation-task.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            exit(0)
+          else:
+            foundRepositoryURL = payloadAsJson["repository"]["html_url"]
+            if (len(foundRepositoryURL) > 1):
+              print('A clone URL has been detected')
+              if foundRepositoryURL == eventOrigin:
+                print('All is well, the repository URL matches and so does the signature. Validate task PASSED')
+                exit(0)
+              else:
+                print('Wanted origin is')
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/validation-task.yaml]: User specified origin is

File: webhooks-extension/config/triggers_prototype/validation-task.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                print(eventOrigin)
+                print('Found clone URL does not match the wanted origin')
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/validation-task.yaml]: Clone URL in event payload does not match the user specified origin

**Comment by a-roberts** [comment]
Closing this (at least for now) as we're looking into using the validation service/interceptor instead of a validate task (https://github.com/tektoncd/experimental/issues/245 is now free for anyone to pick up)

**Pull request closed without merging**